### ISSUE TITLE
Don't suggest adding fulltext indexes unless necessary

### ISSUE THREAD

**Issue by mnelson4** [mnelson4/not-another-related-posts-plugin#3]
Currently, even if you don't have YARPP use title and bodies, it gives you a warning if you are using innoDB database engine, because it doesn't support [fulltext indexes](https://dev.mysql.com/doc/refman/5.5/en/fulltext-search.html).

Understandably, [this notice can cause some concern](https://wordpress.org/support/topic/is-the-plugin-closed-permanently/page/2/#post-11322666). (Unnecessary concern when we aren't even going to use those indexes!)

Plus, adding these indexes can slowdown writes to those important tables. So it's worth measuring considering if the cost-to-benefits (eg how much better  are the related results with it on than off, how does it affect READS, and how does it affect WRITES to those tables, etc).

**Comment by derpyherp** [comment]
I get this warning even though my wp_posts table already uses MyISAM engine (always has as far as I understand):

![](https://i.imgur.com/BoOZ7jg.png)

The notice claims it's using "1" engine. 

Also, creating fulltext indexes through the plugin fails with the message:

> Trying to force fulltext indexing on your table resulted in an error. Your posts table does not have fulltext indexing capabilities.

I used the plugin with "consider titles" option on since many older versions. Don't believe anything changed on the site itself.

**Comment by mnelson4** [comment]
hi @derpyherp, thanks for the error report. Could you please create a separate issue for it? Could you also include the message that says its using engine "1", and maybe a screenshot or text showing what indexes exist on your `wp_posts` table? That might help identify the issue...

**Comment by mnelson4** [comment]
Oh hey @derpyherp, [someone else seem's to have reported the same issue](https://wordpress.org/support/topic/cant-use-indexing/#post-11527075) and I was kinda able to reproduce. You can try https://github.com/mnelson4/yet-another-related-posts-plugin/archive/BUG/add-indeces-when-already-exist.zip, go to settings -> YARPP, then in the pop-up, click "Create fulltext indices", and the message should go away.

Note: I think you may have got that problem because if the database query to check for the fulltext indices can sometimes be slow, so it may fail 1/10000 times or so. But if you're that unlucky user, it will then save a database option saying the indices don't exist *even though they do*. Then, when it tries to recreate them, it of course has an error because they already exist.
My modified version gets around that by simply first double-checking the indices don't actually exist before trying to make them.

**Comment by mnelson4** [comment]
Yep, that branch fixed the issue.

**Comment by rigoleta** [comment]
Will this modified version of YARP be overwritten when updates are made? Thank you

**Comment by mnelson4** [comment]
Hi @rigoleta, this version of YARPP (on GitHub) has effectively diverged from the version on WordPress.org. So, changes made to this version will NOT be overwritten so long as you continue to use this version. If you switch to the version from WordPress.org, those changes will be lost. Does that make sense?

**Comment by rigoleta** [comment]
Yes, it makes sense. Will this version receive any updates? I needed it because I had the same issues exposed in this post, now they are resolved with this version. Thank you

**Comment by mnelson4** [comment]
This version is mostly only receiving critical bugfixes at the moment, it's not being actively developed. I think the version of WordPress.org is receiving more maintenance (it's backed by a commercial company, ShareAholic). So if you might want to suggest to them to integrate the changes from https://github.com/mnelson4/yet-another-related-posts-plugin/pull/5 into their version of YARPP, then you can use theirs.

**Comment by jmeattle** [comment]
Tracked to be pulled into main plugin here:
https://github.com/shareaholic/wp-yet-another-related-posts-plugin/issues/55
